coins/bitcoin/src/rpc.rs

```diff
@@ -0,0 +1,538 @@
+            .await?;
+        Ok(info)
+    }
+
+    pub async fn lock_unspent(&self, outputs: &[OutPoint]) -> Result<bool> {
+        let outputs: Vec<_> = outputs
+            .iter()
+            .map(|o| serde_json::to_value(JsonOutPoint::from(*o)).unwrap())
+            .collect();
+        let info: bool = self
+            .rpc_call::<bool>("lockunspent", &[false.into(), outputs.into()])
+            .await?;
+        Ok(info)
+    }
+
+    pub async fn unlock_unspent(&self, outputs: &[OutPoint]) -> Result<bool> {
+        let outputs: Vec<_> = outputs
+            .iter()
+            .map(|o| serde_json::to_value(JsonOutPoint::from(*o)).unwrap())
+            .collect();
+        let mut ext_args = [true.into(), outputs.into()];
+        let args = handle_defaults(&mut ext_args, &[true.into(), empty_arr()]);
+        let info: bool = self.rpc_call::<bool>("lockunspent", &args).await?;
+        Ok(info)
+    }
+
+    pub async fn get_fee_per_byte(&self, conf_target: Option<usize>) -> anyhow::Result<u64> {
+        let mut ext_args = [opt_into_json(conf_target)?];
+        let defaults = [into_json(100)?];
+        let args = handle_defaults(&mut ext_args, &defaults);
+        let fee: EstimateSmartFeeResult = self
+            .rpc_call::<EstimateSmartFeeResult>("estimatesmartfee", &args)
+            .await?;
+        Ok(fee.fee_rate.unwrap().to_sat())
+    }
+
+    pub async fn get_transaction(&self, tx_hash: &str) -> anyhow::Result<GetTransactionResult> {
+        let mut ext_args = [into_json(tx_hash)?];
+        let args = handle_defaults(&mut ext_args, &[null()]);
+        let tx = self
+            .rpc_call::<GetTransactionResult>("gettransaction", &args)
+            .await?;
+        Ok(tx)
+    }
+
+    pub async fn get_transactions(
+        &self,
+        tx_hashes: Vec<&str>,
+    ) -> anyhow::Result<Vec<GetTransactionResult>> {
+        let mut transactions = Vec::<GetTransactionResult>::new();
+        for one_tx in tx_hashes.iter() {
+            let mut ext_args = [into_json(one_tx)?];
+            let args = handle_defaults(&mut ext_args, &[null()]);
+            let one_transaction = self
+                .rpc_call::<GetTransactionResult>("gettransaction", &args)
+                .await?;
+            transactions.push(one_transaction);
+        }
+        Ok(transactions)
+    }
+
+    pub async fn get_raw_transactions(
+        &self,
+        tx_hashes: Vec<&str>,
+        block_hash : Option<&BlockHash>,
+    ) -> anyhow::Result<Vec<Transaction>> {
+        let mut transactions = Vec::<Transaction>::new();
+        for one_tx in tx_hashes.iter() {
+            let one_transaction = self.get_raw_transaction(&bitcoin::Txid::from_str(one_tx).unwrap(), None, block_hash).await.unwrap();
+            transactions.push(one_transaction);
+        }
+        Ok(transactions)
+    }
+
+    pub async fn is_confirmed(&self, tx_hash: &str) -> anyhow::Result<bool> {
```

Comment:
Remove. While this is suitably deterministic, it hardcodes the confirmation count to 10. The scanner won't need this and I'd rather not support this.